### PR TITLE
Added protection against players joining after a game has ended or if the game is at capacity

### LINKED ISSUE #54: Prevent players from joining game if has finished or has reached max players
Proposed fix is to modify the `createPlayer()` service to add a check for `game.state === GameState.finished` and `game.players.length > game.settings.maxPlayers`

### PR DESCRIPTION
**Pull request by Alxjlv**
Closes #54

### CODE REVIEW

File: backend/src/services/player.service.ts
Comment (taitfuller) [inline backend/src/services/player.service.ts:22]: Since we're using Modern JavaScript you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)!
 
Try: `const gameMax = game?.settings?.maxPlayers || 40`!

**Review by taitfuller — COMMENTED** [review]
Looks good! Try see if you can do that small change using optional chaining but otherwise looks good to merge!

File: backend/src/services/player.service.ts
Comment (Alxjlv) [reply backend/src/services/player.service.ts]: Done!

**Comment by taitfuller** [comment]
Cool, I've just approved #75 so once this is merged add the MaxPlayers constant & we'll be good to go!

**Review by taitfuller — APPROVED** [review]
Great work!

**Pull request merged**
